utils/hct/hctdb.py

```diff
@@ -7803,6 +7899,47 @@ def build_valrules(self):
             "Invalid use of completed record handle.",
         )
 
+        # Linalg ops
+        self.add_valrule(
+            "Instr.MatVecOpIsUnsignedFlagsAreConst",
+            "MatVec Ops Is Unsigned flag is a constant",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgInterpretationParamAreConst",
+            "Interpretation values are constants",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgInvalidRegisterInterpValue",
+            "From Register Interpretation value not in valid set",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgInvalidMemoryInterpValue",
+            "In Memory Interpolation value not in valid set",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgMatrixShapeParamsAreConst",
+            "Matrix Layout, Dimensions and isTranspose are immediate constants",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgInvalidMatrixLayoutValue",
+            "Matrix Layout for Linalg ops not in valid set",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgMatrixLayoutNotTransposable",
+            "Matrix Layout not transposable",
+        )
+
+        self.add_valrule(
+            "Instr.LinalgNotAnUnsignedType",
+            "Unsigned flag set for signed type",
```

Comment:
I think the validation descriptions/messages aren't ideal.  We could update these in a follow-up maybe.

Ideally, we pass two strings in place of this one.  The first is the description of the rule, usually stated in a positive way, like: `"Unsigned flag corresponding to float typed parameter must be false"`

The next string is used as an error string by validation, and can have arguments, like:
`"Unsigned flag is true when corresponding %select{input|output}0 vector is float"`

If the second string is not supplied (like for these cases), the error message defaults to the description string.

Reply:
For this specific case, just to stay consistent with the error messages for SignedInt and float, should we just have the checking in sema and skip the DXIL Validation?

Reply:
As we discussed, we should still have the validation.

Reply:
Yep, left it there and updated per feedback.